pkg/handler/common/provider/aws_test.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2021 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider_test
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/util/diff"
+
+	v1 "k8c.io/kubermatic/v2/pkg/crd/kubermatic/v1"
+	"k8c.io/kubermatic/v2/pkg/handler/common/provider"
+)
+
+func TestAWSSize(t *testing.T) {
+	tests := []struct {
+		name          string
+		region        string
+		resourceQuota v1.MachineDeploymentVMResourceQuota
+		expectedNames []string
+	}{
+		{
+			name:          "test ARM filtering",
+			region:        "eu-central-1",
+			resourceQuota: genDefaultMachineDeploymentVMResourceQuota(),
+			// t4g.small is a instance with ARM which is filtered out here
+			expectedNames: []string{"t3a.small", "t3.small"},
```

Comment:
What would be good is doing a negative test, e.g. increase the quotas and check that no instance types of `a1`, `c6`, `r6`, `t4g` families are returned (prefix matching). WDYT?

Reply:
> I've removed my invalid remark from before.
> 
> @irozzo-1A PTAL. It looks good to me.

LGTM as well, just a small suggestion to improve the test

Reply:
I though about it, but I wanted to leave this test like this so there could be a test as well here which tests the quotas, bad region, or pricing which are all in this AWSSize func.

But I can change it as well, let me know what you think

Reply:
Yeah, good to cover all this, but I think that in the scope of this PR the most important thing to test is that no arm-based instance types are proposed. Ideally, there should be several cases or tests to cover what you mentioned.

Reply:
ok ill change it to prefix. Later we can take care of the other, bit surprising there are no tests for any of this tbh

Reply:
agree, would be better to increase the test coverage a bit

Reply:
Ahh, but there is a problem. The `AWSSize` model does not contain info about instance type family. 

```
// AWSSize represents a object of AWS size.
// swagger:model AWSSize
type AWSSize struct {
	Name       string  `json:"name"`
	PrettyName string  `json:"pretty_name"`
	Memory     float32 `json:"memory"`
	VCPUs      int     `json:"vcpus"`
	GPUs       int     `json:"gpus"`
	Price      float64 `json:"price"`
}
```

Name and prettyName are just "t3a.small" and "T3A Small", so the model would have to change and this would be a bigger change as we need to add Family to it. I can do it but Ill need to test a bit if it doesnt mess something up. @maciaszczykm for UI it wont matter right?

Reply:
What I would do is just simple prefix matching on `Name` at the moment

Reply:
Ok let me see for names, I guess they will all have this `g` in the name

Reply:
Everything that UI doesn't use does not matter. However, we have to remember that this produces some additional network usage so usually we don't add too many things that we are not using. One property shouldn't change much.

Reply:
nvm disregard long comment up, i mixed up some fields

Reply:
Something like this to give you the idea
```
armFamilies := []map[string]struct{}{"a1", "c6", "r6", "tg4"}
for name := range names {
  family := strings.Split(name, ".")[0]
  if _, ok := armFamilies[family] {
    //fail test
  }
}
```

Reply:
didnt notice above before sorry. Changed the test and listed out the ARM instances so its clear